libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs

```diff
@@ -79,10 +34,10 @@ public override async Task OnTurnAsync(ITurnContext turnContext, CancellationTok
                 throw new ArgumentException($"{nameof(turnContext)}.Activity must have non-null Type.");
             }
 
-            if (turnContext.Activity.ChannelId == Channels.Msteams)
-            {
-                _teamsRosterClient = new TeamsRosterClient((ConnectorClient)turnContext.TurnState.Get<IConnectorClient>());
-            }
+            TeamsInfo = new TeamsInfo(
+                turnContext.Activity?.Conversation?.Id,
+                (ConnectorClient)turnContext.TurnState.Get<IConnectorClient>(),
+                turnContext.Activity.GetChannelData<TeamsChannelData>());
```

Comment:
better to say...
  turnContext.Activity.GetChannelData<TeamsChannelData>()?.Team?.Id

i think we only need teamId